src/OrchardCore.Modules/OrchardCore.ContentFields/Indexing/ContentPickerFieldIndexHandler.cs

```diff
@@ -0,0 +1,17 @@
+using System.Threading.Tasks;
+using OrchardCore.ContentFields.Fields;
+using OrchardCore.Indexing;
+
+namespace OrchardCore.ContentFields.Indexing
+{
+    public class ContentPickerFieldIndexHandler : ContentFieldIndexHandler<ContentPickerField>
+    {
+        public override Task BuildIndexAsync(ContentPickerField field, BuildFieldIndexContext context)
+        {
+            var options = context.Settings.ToOptions();
+            context.DocumentIndex.Entries.Add(context.Key, new DocumentIndex.DocumentIndexEntry(string.Join(";", field.ContentItemIds), DocumentIndex.Types.Text, options));
```

Comment:
Don't join the value here, just do a loop on all values and add them every time. The same field property will then have multiple values, and the lucene queries will be able to handle that.

Reply:
fixed